### PR TITLE
Bugfix: lock screen crashes (NMA-1135)

### PR DESCRIPTION
**Pull request by HashEngineering**
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [ ] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/ui/dialogs/AdaptiveDialog.kt [hunk 1]
```diff
@@ -104,6 +104,8 @@ class AdaptiveDialog(@LayoutRes private val layout: Int): DialogFragment() {
     private var onResultListener: ((Boolean?) -> Unit)? = null
     var isMessageSelectable = false
 
+    constructor() : this(R.layout.dialog_simple)
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/ui/dialogs/AdaptiveDialog.kt:107]: The goal is to fix this crash:

```
Fatal Exception: java.lang.RuntimeException
Unable to start activity ComponentInfo{hashengineering.darkcoin.wallet/de.schildbach.wallet.ui.WalletActivity}: 
   androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment
      org.dash.wallet.common.ui.dialogs.AdaptiveDialog: could not find Fragment constructor
```

File: wallet/src/de/schildbach/wallet/ui/LockScreenActivity.kt [hunk 2]
```diff
@@ -529,10 +534,13 @@ open class LockScreenActivity : SecureActivity() {
...
-                    fragment.dismissAllowingStateLoss()
-                } else if (fragment is NavHostFragment) {
-                    dismissDialogFragments(fragment.childFragmentManager)
+                // check to see if the activity is valid and fragment is added to its activity
+                if (fragment.activity != null && fragment.isAdded) {
+                    if (fragment is DialogFragment) {
+                        fragment.dismissAllowingStateLoss()
+                    } else if (fragment is NavHostFragment) {
+                        dismissDialogFragments(fragment.childFragmentManager)
+                    }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/LockScreenActivity.kt:543]: The goal is to fix this crash:

```
Fatal Exception: java.lang.IllegalStateException
Fragment ReceiveDetailsDialog{5fb3ac3} (f9324875-6305-4670-8f78-6a1eb97ebb46) not associated with a fragment manager.
  androidx.fragment.app.Fragment.getParentFragmentManager (Fragment.java:1040)
  com.google.android.material.bottomsheet.BottomSheetDialogFragment.dismissAllowingStateLoss (BottomSheetDialogFragment.java:54)
  de.schildbach.wallet.ui.LockScreenActivity.dismissDialogFragments (LockScreenActivity.kt:518)
  de.schildbach.wallet.ui.LockScreenActivity.onLockScreenActivated (LockScreenActivity.kt:510)
  de.schildbach.wallet.ui.LockScreenActivity.onLogoutListener$lambda-1 (LockScreenActivity.kt:141)
  de.schildbach.wallet.ui.LockScreenActivity.lambda$eS7hrU2lCra4Z7LuRF5JiXGFef0 (LockScreenActivity.kt)
  de.schildbach.wallet.ui.-$$Lambda$LockScreenActivity$eS7hrU2lCra4Z7LuRF5JiXGFef0.onLogout (lambda:2)
  de.schildbach.wallet.AutoLogout$2.run (AutoLogout.java:92) 
```

File: wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java [hunk 3]
```diff
@@ -28,7 +26,7 @@ public class PinRetryController {
     private final static String PREFS_FAILED_PINS = "failed_pins";
     private final static int RETRY_FAIL_TOLERANCE = 3;
     private final static int POW_LOCK_TIME_BASE = 6;
-    private final static int FAIL_LIMIT = 8;
+    private final static int FAIL_LIMIT = 3;
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/preference/PinRetryController.java:29]: This is only for testing purposes.

File: wallet/src/de/schildbach/wallet/ui/EncryptNewKeyChainDialogFragment.java [hunk 4]
```diff
@@ -109,7 +114,10 @@ protected void onBadPassword() {
                 unlockButton.setEnabled(true);
                 unlockButton.setText(getText(R.string.wallet_lock_unlock));
                 pinView.setEnabled(true);
-                pinRetryController.failedAttempt(password);
+                if(pinRetryController.failedAttempt(password)) {
+                    restartService.performRestart(getActivity(), true);
+                    dismiss();
+                }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/EncryptNewKeyChainDialogFragment.java:120]: In theory this dialog is not used anymore due to the changes with to resolve the Non-encrypted Wallets crashes and errors.

We can get rid of it in a new story.

File: wallet/src/de/schildbach/wallet/ui/ResetWalletDialog.kt [hunk 5]
```diff
@@ -46,8 +52,9 @@ class ResetWalletDialog : DialogFragment() {
                     // 2. start OnboardingActivity
                     // 3. close the backstack (Home->More->Security)
                     WalletApplication.getInstance().triggerWipe(context)
-                    startActivity(OnboardingActivity.createIntent(requireContext()))
-                    activity?.finishAffinity()
+                    restartService.performRestart(requireActivity(), true)
+                    //startActivity(OnboardingActivity.createIntent(requireContext()))
+                    //activity?.finishAffinity()
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/ResetWalletDialog.kt:57]: This needs to be removed

File: wallet/src/de/schildbach/wallet/ui/EncryptNewKeyChainDialogFragment.java (hunk 4 above)
Comment (HashEngineering) [reply wallet/src/de/schildbach/wallet/ui/EncryptNewKeyChainDialogFragment.java:120]: Nevermind, this one is still used in very rare cases.

File: wallet/src/de/schildbach/wallet/ui/EncryptKeysDialogFragment.java [hunk 6]
```diff
@@ -319,7 +324,10 @@ public void run() {
                                     log.info("wallet successfully decrypted");
                                 } catch (final KeyCrypterException x) {
                                     log.info("wallet decryption failed: " + x.getMessage());
-                                    pinRetryController.failedAttempt(oldPassword);
+                                    if(pinRetryController.failedAttempt(oldPassword)) {
+                                        restartService.performRestart(getActivity(), true);
+                                        dismiss();
+                                    }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/EncryptKeysDialogFragment.java:330]: In theory this dialog is not used anymore due to the changes with to resolve the Non-encrypted Wallets crashes and errors.

We can get rid of it in a new story.

This dialog was the old Set Pin dialog.

File: wallet/src/de/schildbach/wallet/ui/AppUpgradeActivity.kt [hunk 7]
```diff
@@ -34,6 +35,7 @@ import kotlinx.android.synthetic.main.activity_app_update.*
 import org.dash.wallet.common.Configuration
 import java.util.concurrent.TimeUnit
 
+@AndroidEntryPoint
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/AppUpgradeActivity.kt:38]: QA found this crash when upgrading from a v5 wallet to v7.

```
AndroidRuntime: Caused by: java.lang.IllegalStateException: Hilt Fragments must be attached to an @AndroidEntryPoint Activity. Found: class de.schildbach.wallet.ui.AppUpgradeActivity
```

**Review by Syn-McJ — APPROVED** [review]

**Pull request merged**